### PR TITLE
[WD: 48] Create module for Wishes Filter

### PR DESCRIPTION
**Pull request by christinasugiono**

### CODE REVIEW

File: app/controllers/organizations_controller.rb [hunk 1]
```diff
@@ -71,10 +71,40 @@ def feedback
     @wishlists = @organization.wishlists
     authorize @organization
     @wish = Wish.new
-    @wishes = @organization.wishes.order(created_at: :desc)
-    return unless params[:filter].present?
-
-    @wishes = Wishes::FeedbackFilterer.new(filter_params:, scope: @wishes).call
+    @wishes = @organization.wishes
```
Comment (Danrod16) [inline app/controllers/organizations_controller.rb:74]: Do we need all these comments?

Comment (Danrod16) [reply app/controllers/organizations_controller.rb:74]: Also why do we need this line if we assign @wishes right after without a conditional?

File: app/services/wishes/feedback_filterer.rb [hunk 2]
```diff
@@ -32,20 +38,27 @@ def apply_stage_filter
     end
 
     def apply_wishlist_filter
-      return unless wishlist_id.present?
+      return unless board.present?
+
```
Comment (Danrod16) [inline app/services/wishes/feedback_filterer.rb:42]: No need for space here :)

File: app/services/wishes/feedback_filterer.rb [hunk 3]
```diff
@@ -32,20 +38,27 @@ def apply_stage_filter
...
+      return unless start_date.present? || end_date.present?
 
-      wishlist = Wishlist.find_by(id: wishlist_id)
-      @scope = scope.where(wishlist:) if wishlist
+      @scope = scope.where(created_at: (start_date.to_date)...(end_date.to_date + 1.day))
+    end
+
+    def apply_query_filter
+      return unless query.present?
+
```
Comment (Danrod16) [inline app/services/wishes/feedback_filterer.rb:54]: No need for space here

File: app/services/wishes/feedback_filterer.rb [hunk 4]
```diff
@@ -32,20 +38,27 @@ def apply_stage_filter
...
+      @scope = scope.search_by_title_and_description(query)
     end
 
     def apply_order
       return unless order_column.present?
```
Comment (Danrod16) [inline app/services/wishes/feedback_filterer.rb:60]: And here

File: app/views/wishes/_filter.html.erb [hunk 5]
```diff
@@ -9,35 +9,39 @@
...
-          <%= image_tag 'sort.png', alt: "sort-icon", class: ("active" if params[:sort_by]) %>
+          <%= image_tag 'sort.png', alt: "sort-icon", class: ("active" if params[:order_column]) %>
         </button>
         <ul class="dropdown-menu">
           <li class="dropdown-header">SORT BY</li>
           <li><hr class="dropdown-divider"></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Newest" %>"><%= link_to "Newest", request.params.merge(sort_by: "Newest") %></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Oldest" %>"><%= link_to "Oldest", request.params.merge(sort_by: "Oldest") %></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Votes" %>"><%= link_to "Votes", request.params.merge(sort_by: "Votes") %></li>
+          <li class="dropdown-item <%= "active" if params[:order_column] == "created_at" && params[:order_direction] == "DESC" %>"><%= link_to "Newest", request.params.merge(order_column: "created_at", order_direction: "DESC") %></li>
```
Comment (Danrod16) [inline app/views/wishes/_filter.html.erb:17]: This code can be refactored like so:
In the Application helper for example:
```
def sortable_link(text, column, direction)
    active = params[:order_column] == column && params[:order_direction] == direction
    link_to(text, request.params.merge(order_column: column, order_direction: direction), 
    class: "dropdown-item #{'active' if active}")
end
```

In the view:

```
<ul class="dropdown-menu">
  <li class="dropdown-header">SORT BY</li>
  <li><hr class="dropdown-divider"></li>
  <li><%= sortable_link('Newest', 'created_at', 'DESC') %></li>
  <li><%= sortable_link('Oldest', 'created_at', 'ASC') %></li>
  <li><%= sortable_link('Votes', 'votes_count', 'DESC') %></li>
</ul>
```

WDYT?

File: app/views/wishes/_filter.html.erb [hunk 6]
```diff
@@ -9,35 +9,39 @@
...
+          <li class="dropdown-item <%= "active" if params[:order_column] == "created_at" && params[:order_direction] == "ASC" %>"><%= link_to "Oldest", request.params.merge(order_column: "created_at", order_direction: "ASC") %></li>
+          <li class="dropdown-item <%= "active" if params[:order_column] == "votes_count" && params[:order_direction] == "DESC" %>"><%= link_to "Votes", request.params.merge(order_column: "votes_count", order_direction: "DESC") %></li>
         </ul>
       </div>
       <%# Filter %>
       <div class="dropdown">
         <button class="p-0" type="button" data-bs-toggle="dropdown" aria-expanded="false">
-          <%= image_tag 'filter.png', alt: "filter-icon", class: ("active" if params[:filter_by]) %>
-          <% if params[:filter_by] %>
+          <%= image_tag 'filter.png', alt: "filter-icon", class: ("active" if params[:board].present? || params[:start_date].present? || params[:end_date].present?) %>
```
Comment (Danrod16) [inline app/views/wishes/_filter.html.erb:25]: Something similar here :)

**Review by Danrod16 — CHANGES_REQUESTED** [review]

File: app/views/wishes/_filter.html.erb (hunk 5 above)
Comment (christinasugiono) [reply app/views/wishes/_filter.html.erb:17]: I will try this!

File: app/services/wishes/feedback_filterer.rb (hunk 3 above)
Comment (christinasugiono) [reply app/services/wishes/feedback_filterer.rb:54]: rubocop ask for space after return keywords :')

File: app/views/wishes/_filter.html.erb [hunk 7]
```diff
@@ -2,67 +2,69 @@
...
                                       value: params[:end_date] %>
                   </li>
+                  <%= f.submit 'Filter', class:"d-none" %>
               <% end %>
             </ul>
           </li>
-          <li class="dropdown-item <%= "active" if params[:filter_by] == "Board" %>">
-            <a href="#">
+          <li class="dropdown-item">
+            <a href="#" class="<%= "active" if params[:board].present? %>">
```
Comment (Danrod16) [inline app/views/wishes/_filter.html.erb:58]: Can we change a tags to link to tags for consistency, please 🙏

**Review by Danrod16 — CHANGES_REQUESTED** [review]

**Pull request merged**